packages/Caching/Config/FileHashComputer.php

```diff
@@ -21,14 +16,8 @@ public function compute(string $filePath): string
     {
         $this->ensureIsPhp($filePath);
 
-        $containerBuilder = new ContainerBuilder();
-
-        $fileLoader = $this->createFileLoader($filePath, $containerBuilder);
-        $fileLoader->load($filePath);
-
-        $parameterBag = $containerBuilder->getParameterBag();
-
-        return $this->arrayToHash($containerBuilder->getDefinitions()) . $this->arrayToHash($parameterBag->all());
+        $parametersHash = SimpleParameterProvider::hash();
+        return sha1($filePath . $parametersHash);
```

Comment:
creating a hash on something which is already hashed increases possibility of hash collisions.

maybe instead add a parameter to `SimpleParameterProvider::hash()` which allows to give additional information from the outside which will be added to the parameters internal hashing.

something like

```suggestion
        return SimpleParameterProvider::hash($filePath);
```

Reply:
I see. The `SimpleParameterProvider::hash()` should only know about itself, nothing else. Similar way the previous implementation dumped unique hash for symfony container with current parameter/service setup.